tests/EStimLibrary.UnitTests/Core/Stimulation/ThreadConfigDataPerStimulatorTest.cs

```diff
@@ -0,0 +1,269 @@
+using EStimLibrary.Core;
+using EStimLibrary.Core.Stimulation;
+using EStimLibrary.Core.HardwareInterfaces;
+
+
+namespace EStimLibrary.UnitTests.Core.Stimulation;
+
+/// <summary>
+/// Unit tests for the ThreadConfigDataPerStimulator record.
+/// </summary>
+public class ThreadConfigDataPerStimulatorTest
```

Comment:
Overall, nice naming and doc-commenting of methods! Can you add a doc-comment for the source class itself, too?